build_test.go

```diff
@@ -0,0 +1,300 @@
+package phpstart_test
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+	phpstart "github.com/paketo-buildpacks/php-start"
+	"github.com/paketo-buildpacks/php-start/fakes"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+)
+
+func testBuild(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect = NewWithT(t).Expect
+
+		layersDir  string
+		workingDir string
+		cnbDir     string
+
+		buffer  *bytes.Buffer
+		procMgr *fakes.ProcMgr
+
+		build packit.BuildFunc
+	)
+
+	it.Before(func() {
+		var err error
+		layersDir, err = os.MkdirTemp("", "layers")
+		Expect(err).NotTo(HaveOccurred())
+
+		cnbDir, err = os.MkdirTemp("", "cnb")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(os.Mkdir(filepath.Join(cnbDir, "bin"), 0700)).To(Succeed())
+		Expect(os.WriteFile(filepath.Join(cnbDir, "bin", "procmgr-binary"), []byte{}, 0644)).To(Succeed())
+
+		workingDir, err = os.MkdirTemp("", "working-dir")
+		Expect(err).NotTo(HaveOccurred())
+
+		buffer = bytes.NewBuffer(nil)
+		logEmitter := scribe.NewEmitter(buffer)
+
+		procMgr = &fakes.ProcMgr{}
+		build = phpstart.Build(procMgr, logEmitter)
+	})
+
+	it.After(func() {
+		Expect(os.RemoveAll(layersDir)).To(Succeed())
+		Expect(os.RemoveAll(cnbDir)).To(Succeed())
+		Expect(os.RemoveAll(workingDir)).To(Succeed())
+	})
+
+	context("the PHP_HTTPD_PATH env var is set", func() {
```

Comment:
I think I understand why this case is here, but just to validate my own understanding:
Am I correct in thinking this case would be an unlikely one given `PHPRC`, `PHP_FPM_PATH` and `PHP_HTTPD_PATH` would have been by previous buildpacks?

Reply:
Yes, you're right on. This won't happen as of now with the way the buildpacks are set up. The plan is to eventually (post re-write probably) support FPM running in its own container, in which case this situation will become more common.

Reply:
Understood. The `php-fpm` buildpack is expected to be optional based on [RFC 0003](https://github.com/paketo-buildpacks/rfcs/blob/f79658839d4fe3fb3f4795f68cd63e8b8f094d70/text/php/0003-php-start.md?plain=1#L92), but it doesn't appear to be in this case. Is that also related to the plan stated above?